packages/ts-client/src/FishjamClient.ts

```diff
@@ -314,11 +315,15 @@ export class FishjamClient<PeerMetadata, TrackMetadata> extends (EventEmitter as
    *
    * @param {ConnectConfig} config - Configuration object for the client
    */
-  connect(config: ConnectConfig<PeerMetadata>): void {
+  public async connect(config: ConnectConfig<PeerMetadata>): Promise<void> {
+    const result = new ConnectPromise(this);
+
     this.reconnectManager.reset(config.peerMetadata);
     this.connectConfig = config;
 
     this.initConnection(config.peerMetadata);
+
+    return result.getPromise();
```

Comment:
with above change, you can also simplify calls here:
```ts
    const result = connectPromiseFun(this);
```
and
```suggestion
    return result;
```
(or `return await result` to be more explicit)